avocado/core/suite.py

```diff
@@ -0,0 +1,141 @@
+from enum import Enum
+from uuid import uuid1
+
+from .exceptions import OptionValidationError
+from .future.settings import settings
+from .loader import loader, LoaderError, LoaderUnhandledReferenceError
+from .resolver import resolve
+from .tags import filter_test_tags
+from .test import DryRunTest
+from .utils import resolutions_to_tasks
+from .varianter import Varianter
+
+
+class TestSuiteError(Exception):
+    pass
+
+
+class TestSuiteStatus(Enum):
+    RESOLUTION_NOT_STARTED = 0
```

Comment:
Nitpick: when we don't have a reason for manually settings values, such as mapping external interfaces, I find it convenient to just use an `object()` to guarantee uniqueness (because `auto` is only available on 3.6 and later https://docs.python.org/3/library/enum.html#enum.auto).

Reply:
I like the numbers because sometimes it is easy to debug when the number it is showing instead of the string.